recipes/libelfin/all/CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+project(${CONAN_PACKAGE_NAME} VERSION ${CONAN_PACKAGE_VERSION})
+conan_basic_setup()
+
+file(GLOB_RECURSE elf_sources source_subfolder/elf/*.cc)
+set(elf_headers
+    source_subfolder/elf/common.hh
+    source_subfolder/elf/data.hh
+    source_subfolder/elf/elf++.hh)
+file(GLOB_RECURSE dwarf_sources source_subfolder/dwarf/*.cc)
+set(dwarf_headers
+    source_subfolder/dwarf/data.hh
+    source_subfolder/dwarf/dwarf++.hh
+    source_subfolder/dwarf/small_vector.hh)
+
+add_custom_command(
+    OUTPUT source_subfolder/elf/to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo '// Automatically generated' > to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo '// DO NOT EDIT' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo '\#include \"data.hh\"' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo '\#include \"to_hex.hh\"' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo 'ELFPP_BEGIN_NAMESPACE' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+    COMMAND python3 enum-print.py -u --hex --no-type --mask shf --mask pf -x loos -x hios -x loproc -x hiproc < data.hh >> to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo 'ELFPP_END_NAMESPACE' >> to_string.cc
+    DEPENDS source_subfolder/elf/enum-print.py source_subfolder/elf/data.hh
+    WORKING_DIRECTORY source_subfolder/elf)
+
+add_custom_command(
+    OUTPUT source_subfolder/dwarf/to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo '// Automatically generated' > to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo '// DO NOT EDIT' >> to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo '\#include \"internal.hh\"' >> to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo 'DWARFPP_BEGIN_NAMESPACE' >> to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+    COMMAND python3 ../elf/enum-print.py < dwarf++.hh >> to_string.cc
+    COMMAND python3 ../elf/enum-print.py -s _ -u --hex -x hi_user -x lo_user < data.hh >> to_string.cc
+	COMMAND ${CMAKE_COMMAND} -E echo 'DWARFPP_END_NAMESPACE' >> to_string.cc
+    DEPENDS source_subfolder/elf/enum-print.py source_subfolder/dwarf/data.hh
+    WORKING_DIRECTORY source_subfolder/dwarf)
+
+add_library(elf++ ${elf_sources} source_subfolder/elf/to_string.cc)
+set_target_properties(elf++ PROPERTIES
+    PUBLIC_HEADER "${elf_headers}"
+    VERSION ${CMAKE_PROJECT_VERSION})
+
+add_library(dwarf++ ${dwarf_sources} source_subfolder/dwarf/to_string.cc)
+set_target_properties(dwarf++ PROPERTIES
+    PUBLIC_HEADER "${dwarf_headers}"
+    VERSION ${CMAKE_PROJECT_VERSION})
+
+install(TARGETS elf++
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    PUBLIC_HEADER DESTINATION include/libelfin/elf)
+install(TARGETS dwarf++
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    PUBLIC_HEADER DESTINATION include/libelfin/dwarf)
```

Comment:
These miss a `RUNTIME DESTINATION bin` to put the dll for Windows.
Does this package support MSVC dll's?

Reply:
Fixed.
This is a pure library package, I don't see why it wouldn't support dll's.

Reply:
shared libraries on windows uses dll+import libraries. And to export some symbol from a dll (and thus put it in the import library), you need to tag those symbols using `__declspec(dllexport)` when exporting and `__declspec(dllimport)` when using the dll. When using the static library, then NO tags should be supplied.